src/Game/Games/CookieClicker/Controler/ServerCookieController.java

```diff
@@ -102,7 +102,9 @@ private boolean validatePressedKey(ActionEnum actionEnum) {
         @Override
         public void update(Observable o, Object arg) {
             MessageDataObject messageDataObject = (MessageDataObject) arg;
-            ServerCookieController.this.play((ActionEnum) messageDataObject.getData());
+            if (messageDataObject.getData() instanceof ActionEnum) {
```

Comment:
I was thinking you could test the value returned by `messageDataObject.getType()` instead

Reply:
The both are correct but I'll send a commit to use the type instead of the value